app/models/cancellation_reason.rb

```diff
@@ -15,4 +19,14 @@ def self.unwanted
   def self.donor_cancelled
     find_by(name_en: "Donor cancelled")
   end
+
+  def self.cancellation_reasons_by(options)
+    if options[:ids]
+      where(id: options[:ids].split(",").flatten)
```

Comment:
I know you've left this in as it was there before but I don't think ember actually uses `params[:ids]` so it can be dropped. This would simplify the method to something like

```
def cancellation_reasons_for(for)
  return where() unless CANCELLATION_REASONS_TYPE.include(for)
  send("visible_to_#{for}")
end
```